django/db/migrations/state.py

```diff
@@ -403,6 +410,12 @@ def from_model(cls, model, exclude_rels=False):
                 elif name == "index_together":
                     it = model._meta.original_attrs["index_together"]
                     options[name] = set(normalize_together(it))
+                elif name == "indexes":
+                    indexes = model._meta.original_attrs["indexes"]
+                    for index in indexes:
+                        # Set index._name of all indexes by calling index.name
+                        index.name
```

Comment:
@MarkusH, wanted to flag this for your review. It doesn't look good to have attribute access have the side effect of setting some other attribute on the class. Agreed?

@akki, Remind me why the name computation can't be done in `Index.__init__()`?